File: backend/pyfactor/core/business_types.py
"""
Unified Business Types Configuration
This is the single source of truth for all business types across the application.
Used by: PlaceholderBusiness, BusinessListing, UserProfile, BusinessDetails
"""

# Comprehensive business type choices - used everywhere
BUSINESS_TYPE_CHOICES = [
    # Service Businesses (Jobs only)
    ('HOME_SERVICES', 'Home Services (Plumbing, Electrical, HVAC)'),
    ('CONSTRUCTION', 'Construction & Contracting'),
    ('CLEANING', 'Cleaning & Maintenance'),
    ('AUTOMOTIVE_REPAIR', 'Automotive Repair'),
    ('PROFESSIONAL_SERVICES', 'Professional Services (Consulting, Accounting, Legal)'),
    ('CREATIVE_SERVICES', 'Creative Services (Design, Photography)'),
    ('EDUCATION_TRAINING', 'Education & Training Centers'),
    ('EVENT_PLANNING', 'Event Planning & Catering'),
    ('SECURITY_SERVICES', 'Security Services'),
    
    # Retail Businesses (POS only)
    ('RETAIL_STORE', 'Retail Store'),
    ('RESTAURANT_CAFE', 'Restaurant/Cafe'),
    ('GROCERY_MARKET', 'Grocery/Market'),
    ('PHARMACY', 'Pharmacy'),
    ('ONLINE_STORE', 'Online Store'),
    ('FASHION_CLOTHING', 'Fashion & Clothing'),
    ('ELECTRONICS_TECH', 'Electronics & Technology'),
    ('HARDWARE_BUILDING', 'Hardware & Building Supplies'),
    ('BOOKSTORE_STATIONERY', 'Bookstore & Stationery'),
    ('FUEL_STATION', 'Fuel/Petrol Station'),
    
    # Mixed Businesses (Both Jobs & POS)
    ('SALON_SPA', 'Salon/Spa'),
    ('MEDICAL_DENTAL', 'Medical/Dental Practice'),
    ('VETERINARY', 'Veterinary Clinic'),
    ('FITNESS_CENTER', 'Fitness Center'),
    ('AUTO_PARTS_REPAIR', 'Auto Parts & Repair'),
    ('WAREHOUSE_STORAGE', 'Warehouse/Storage'),
    ('MANUFACTURING', 'Manufacturing'),
    ('HOTEL_HOSPITALITY', 'Hotel & Hospitality'),
    ('TECH_SERVICES', 'Technology Services & Repair'),
    ('PRINT_MEDIA', 'Printing & Media Services'),
    ('MOBILE_MONEY', 'Mobile Money & Transfers'),
    
    # Other (Both features)
    ('LOGISTICS_FREIGHT', 'Logistics & Freight'),
    ('FINANCIAL_SERVICES', 'Financial Services (Banks, Insurance)'),
    ('REAL_ESTATE', 'Real Estate'),
    ('AGRICULTURE', 'Agriculture'),
    ('NON_PROFIT', 'Non-Profit'),
    ('TRANSPORT_SERVICE', 'Transport Services (Taxi, Bus, Airlines)'),
    ('MINING_ENERGY', 'Mining & Energy'),
    ('TOURISM_TRAVEL', 'Tourism & Travel Agencies'),
    ('IMPORT_EXPORT', 'Import/Export & Trading'),
    ('TELECOM', 'Telecommunications'),
    ('OTHER', 'Other'),
]

# Business categories for feature access control
BUSINESS_CATEGORIES = {
    'SERVICE': {
        'label': 'Service Businesses',
        'features': ['jobs'],
        'types': [
            'HOME_SERVICES',
            'CONSTRUCTION',
            'CLEANING',
            'AUTOMOTIVE_REPAIR',
            'PROFESSIONAL_SERVICES',
            'CREATIVE_SERVICES',
            'EDUCATION_TRAINING',
            'EVENT_PLANNING',
            'SECURITY_SERVICES',
        ]
    },
    'RETAIL': {
        'label': 'Retail Businesses',
        'features': ['pos'],
        'types': [
            'RETAIL_STORE',
            'RESTAURANT_CAFE',
            'GROCERY_MARKET',
            'PHARMACY',
            'ONLINE_STORE',
            'FASHION_CLOTHING',
            'ELECTRONICS_TECH',
            'HARDWARE_BUILDING',
            'BOOKSTORE_STATIONERY',
            'FUEL_STATION',
        ]
    },
    'MIXED': {
        'label': 'Mixed Businesses',
        'features': ['jobs', 'pos'],
        'types': [
            'SALON_SPA',
            'MEDICAL_DENTAL',
            'VETERINARY',
            'FITNESS_CENTER',
            'AUTO_PARTS_REPAIR',
            'WAREHOUSE_STORAGE',
            'MANUFACTURING',
            'HOTEL_HOSPITALITY',
            'TECH_SERVICES',
            'PRINT_MEDIA',
            'MOBILE_MONEY',
        ]
    },
    'OTHER': {
        'label': 'Other',
        'features': ['jobs', 'pos'],
        'types': [
            'LOGISTICS_FREIGHT',
            'FINANCIAL_SERVICES',
            'REAL_ESTATE',
            'AGRICULTURE',
            'NON_PROFIT',
            'TRANSPORT_SERVICE',
            'MINING_ENERGY',
            'TOURISM_TRAVEL',
            'IMPORT_EXPORT',
            'TELECOM',
            'OTHER',
        ]
    }
}

def get_features_for_business_type(business_type):
    """Get enabled features for a business type"""
    if not business_type:
        return ['jobs', 'pos']  # Default to all features
    
    for category, config in BUSINESS_CATEGORIES.items():
        if business_type in config['types']:
            return config['features']
    
    return ['jobs', 'pos']  # Default to all features

def get_category_for_business_type(business_type):
    """Get category (SERVICE/RETAIL/MIXED/OTHER) for a business type"""
    if not business_type:
        return 'OTHER'
    
    for category, config in BUSINESS_CATEGORIES.items():
        if business_type in config['types']:
            return category
    
    return 'OTHER'

def get_simplified_business_type(business_type):
    """Compatibility function - returns the business type itself"""
    valid_types = [choice[0] for choice in BUSINESS_TYPE_CHOICES]
    return business_type if business_type in valid_types else 'OTHER'

# Mapping from old categories to new business types
# Used for migrating existing data
OLD_CATEGORY_MAPPING = {
    # Healthcare related
    'healthcare': 'MEDICAL_DENTAL',
    'hospital': 'MEDICAL_DENTAL',
    'clinic': 'MEDICAL_DENTAL',
    'medical': 'MEDICAL_DENTAL',
    'pharmacy': 'PHARMACY',
    
    # Financial services
    'banking': 'FINANCIAL_SERVICES',
    'bank': 'FINANCIAL_SERVICES',
    'financial_services': 'FINANCIAL_SERVICES',
    'insurance': 'FINANCIAL_SERVICES',
    'microfinance': 'FINANCIAL_SERVICES',
    
    # Transportation
    'transportation': 'TRANSPORT_SERVICE',
    'transport': 'TRANSPORT_SERVICE',
    'airline': 'TRANSPORT_SERVICE',
    'logistics': 'LOGISTICS_FREIGHT',
    'freight': 'LOGISTICS_FREIGHT',
    'shipping': 'LOGISTICS_FREIGHT',
    
    # Hospitality
    'hotel': 'HOTEL_HOSPITALITY',
    'hospitality': 'HOTEL_HOSPITALITY',
    'lodge': 'HOTEL_HOSPITALITY',
    'guest_house': 'HOTEL_HOSPITALITY',
    
    # Food & Beverage
    'restaurant': 'RESTAURANT_CAFE',
    'cafe': 'RESTAURANT_CAFE',
    'food': 'RESTAURANT_CAFE',
    'food_delivery': 'RESTAURANT_CAFE',
    'beverage': 'RESTAURANT_CAFE',
    
    # Retail
    'retail': 'RETAIL_STORE',
    'store': 'RETAIL_STORE',
    'shop': 'RETAIL_STORE',
    'supermarket': 'GROCERY_MARKET',
    'grocery': 'GROCERY_MARKET',
    'market': 'GROCERY_MARKET',
    
    # Technology
    'technology': 'TECH_SERVICES',
    'tech': 'TECH_SERVICES',
    'it': 'TECH_SERVICES',
    'software': 'TECH_SERVICES',
    'e_commerce': 'ONLINE_STORE',
    'telecom': 'TELECOM',
    'telecommunications': 'TELECOM',
    
    # Real Estate & Construction
    'real_estate': 'REAL_ESTATE',
    'property': 'REAL_ESTATE',
    'construction': 'CONSTRUCTION',
    'building': 'CONSTRUCTION',
    
    # Manufacturing & Industry
    'manufacturing': 'MANUFACTURING',
    'industrial': 'MANUFACTURING',
    'factory': 'MANUFACTURING',
    'production': 'MANUFACTURING',
    
    # Energy & Mining
    'mining': 'MINING_ENERGY',
    'energy': 'MINING_ENERGY',
    'oil': 'MINING_ENERGY',
    'gas': 'MINING_ENERGY',
    'petroleum': 'MINING_ENERGY',
    'solar': 'MINING_ENERGY',
    'utilities': 'MINING_ENERGY',
    
    # Agriculture
    'agriculture': 'AGRICULTURE',
    'farming': 'AGRICULTURE',
    'agribusiness': 'AGRICULTURE',
    
    # Beauty & Personal Care
    'beauty_salon': 'SALON_SPA',
    'salon': 'SALON_SPA',
    'spa': 'SALON_SPA',
    'barber': 'SALON_SPA',
    'cosmetics': 'SALON_SPA',
    
    # Fashion
    'fashion': 'FASHION_CLOTHING',
    'clothing': 'FASHION_CLOTHING',
    'apparel': 'FASHION_CLOTHING',
    'textiles': 'FASHION_CLOTHING',
    
    # Automotive
    'automotive': 'AUTO_PARTS_REPAIR',
    'auto': 'AUTO_PARTS_REPAIR',
    'car': 'AUTO_PARTS_REPAIR',
    'vehicle': 'AUTO_PARTS_REPAIR',
    
    # Professional Services
    'consulting': 'PROFESSIONAL_SERVICES',
    'legal': 'PROFESSIONAL_SERVICES',
    'accounting': 'PROFESSIONAL_SERVICES',
    'professional': 'PROFESSIONAL_SERVICES',
    
    # Education
    'education': 'EDUCATION_TRAINING',
    'school': 'EDUCATION_TRAINING',
    'training': 'EDUCATION_TRAINING',
    'university': 'EDUCATION_TRAINING',
    
    # Media & Entertainment
    'media': 'PRINT_MEDIA',
    'printing': 'PRINT_MEDIA',
    'publishing': 'PRINT_MEDIA',
    'advertising': 'CREATIVE_SERVICES',
    'marketing': 'CREATIVE_SERVICES',
    
    # Tourism
    'tourism': 'TOURISM_TRAVEL',
    'travel': 'TOURISM_TRAVEL',
    'tour': 'TOURISM_TRAVEL',
    
    # Import/Export
    'import': 'IMPORT_EXPORT',
    'export': 'IMPORT_EXPORT',
    'trading': 'IMPORT_EXPORT',
    'wholesale': 'IMPORT_EXPORT',
    
    # Fitness
    'fitness': 'FITNESS_CENTER',
    'gym': 'FITNESS_CENTER',
    'sports': 'FITNESS_CENTER',
    
    # Conglomerate/Mixed
    'conglomerate': 'OTHER',
    'diversified': 'OTHER',
    'holding': 'OTHER',
    
    # Aerospace
    'aerospace': 'MANUFACTURING',
    
    # Event Planning
    'event': 'EVENT_PLANNING',
    'catering': 'EVENT_PLANNING',
    
    # Security
    'security': 'SECURITY_SERVICES',
    
    # Fuel stations
    'fuel': 'FUEL_STATION',
    'petrol': 'FUEL_STATION',
    'gas_station': 'FUEL_STATION',
    
    # Electronics
    'electronics': 'ELECTRONICS_TECH',
    'computers': 'ELECTRONICS_TECH',
    'phones': 'ELECTRONICS_TECH',
    
    # Hardware
    'hardware': 'HARDWARE_BUILDING',
    'building_supplies': 'HARDWARE_BUILDING',
    
    # Books
    'bookstore': 'BOOKSTORE_STATIONERY',
    'stationery': 'BOOKSTORE_STATIONERY',
    
    # Mobile Money
    'mobile_money': 'MOBILE_MONEY',
    'money_transfer': 'MOBILE_MONEY',
    'mpesa': 'MOBILE_MONEY',
    
    # Warehouse
    'warehouse': 'WAREHOUSE_STORAGE',
    'storage': 'WAREHOUSE_STORAGE',
    
    # Veterinary
    'veterinary': 'VETERINARY',
    'animal': 'VETERINARY',
    
    # Non-profit
    'non_profit': 'NON_PROFIT',
    'ngo': 'NON_PROFIT',
    'charity': 'NON_PROFIT',
    
    # Home Services
    'plumbing': 'HOME_SERVICES',
    'electrical': 'HOME_SERVICES',
    'hvac': 'HOME_SERVICES',
    'maintenance': 'HOME_SERVICES',
    
    # Cleaning
    'cleaning': 'CLEANING',
    'laundry': 'CLEANING',
    'dry_cleaning': 'CLEANING',
    
    # Old BusinessListing categories mapping
    'RESTAURANT': 'RESTAURANT_CAFE',
    'CAFE': 'RESTAURANT_CAFE',
    'GROCERY': 'GROCERY_MARKET',
    'PHARMACY': 'PHARMACY',
    'RETAIL': 'RETAIL_STORE',
    'FASHION': 'FASHION_CLOTHING',
    'ELECTRONICS': 'ELECTRONICS_TECH',
    'HOME_GARDEN': 'RETAIL_STORE',
    'BEAUTY': 'SALON_SPA',
    'HEALTH': 'MEDICAL_DENTAL',
    'FITNESS': 'FITNESS_CENTER',
    'AUTOMOTIVE': 'AUTO_PARTS_REPAIR',
    'SERVICES': 'PROFESSIONAL_SERVICES',
    'EDUCATION': 'EDUCATION_TRAINING',
    'ENTERTAINMENT': 'EVENT_PLANNING',
    'OTHER': 'OTHER',
}

def migrate_old_category(old_category):
    """
    Convert old category names to new business type
    Returns the new business type or 'OTHER' if no mapping found
    """
    if not old_category:
        return 'OTHER'
    
    # Check if already a valid business type
    valid_types = [choice[0] for choice in BUSINESS_TYPE_CHOICES]
    if old_category in valid_types:
        return old_category
    
    # Try direct mapping
    old_category_lower = old_category.lower() if isinstance(old_category, str) else ''
    if old_category_lower in OLD_CATEGORY_MAPPING:
        return OLD_CATEGORY_MAPPING[old_category_lower]
    
    # Try uppercase version (for BusinessListing categories)
    if old_category in OLD_CATEGORY_MAPPING:
        return OLD_CATEGORY_MAPPING[old_category]
    
    # Try partial matching
    for key, value in OLD_CATEGORY_MAPPING.items():
        if key in old_category_lower or old_category_lower in key:
            return value
    
    return 'OTHER'

File: mobile/DottAppNative/src/constants/businessTypes.js
// Business Types Configuration for Progressive Registration
// This matches the backend business_types.py structure

export const ENTITY_TYPES = [
  { value: 'INDIVIDUAL', label: 'Individual Service Provider', description: 'Freelancer, Sole Proprietor, Independent Contractor' },
  { value: 'SMALL_BUSINESS', label: 'Small Business', description: '1-10 employees' },
  { value: 'MEDIUM_BUSINESS', label: 'Medium Business', description: '11-50 employees' },
  { value: 'LARGE_COMPANY', label: 'Large Company', description: '50+ employees' },
  { value: 'NON_PROFIT', label: 'Non-Profit Organization', description: 'NGO, Charity, Foundation' },
];

export const INDIVIDUAL_CATEGORIES = {
  'TRANSPORT_DELIVERY': {
    label: 'Transportation & Delivery',
    icon: 'car-outline',
    types: [
      { value: 'COURIER', label: 'Courier/Delivery Driver', requiresVehicle: true },
      { value: 'TAXI_DRIVER', label: 'Taxi/Ride Share Driver', requiresVehicle: true },
      { value: 'TRUCK_DRIVER', label: 'Truck/Freight Driver', requiresVehicle: true },
      { value: 'BODA_BODA', label: 'Boda Boda (Motorcycle Taxi)', requiresVehicle: true },
      { value: 'MOVING_SERVICES', label: 'Moving Services', requiresVehicle: true },
    ]
  },
  'HOME_SERVICES': {
    label: 'Home & Trade Services',
    icon: 'hammer-outline',
    types: [
      { value: 'PLUMBER', label: 'Plumber' },
      { value: 'ELECTRICIAN', label: 'Electrician' },
      { value: 'CARPENTER', label: 'Carpenter' },
      { value: 'PAINTER', label: 'Painter' },
      { value: 'MASON', label: 'Mason/Bricklayer' },
      { value: 'WELDER', label: 'Welder' },
      { value: 'MECHANIC', label: 'Mechanic' },
      { value: 'HVAC_TECH', label: 'HVAC Technician' },
      { value: 'HANDYMAN', label: 'General Handyman' },
      { value: 'CLEANER', label: 'Cleaner/Housekeeper' },
      { value: 'LANDSCAPER', label: 'Landscaper/Gardener' },
      { value: 'PEST_CONTROL', label: 'Pest Control' },
    ]
  },
  'PROFESSIONAL': {
    label: 'Professional Services',
    icon: 'briefcase-outline',
    types: [
      { value: 'CONSULTANT', label: 'Consultant' },
      { value: 'ACCOUNTANT', label: 'Accountant' },
      { value: 'LAWYER', label: 'Lawyer/Legal Advisor' },
      { value: 'REAL_ESTATE_AGENT', label: 'Real Estate Agent' },
      { value: 'INSURANCE_AGENT', label: 'Insurance Agent' },
      { value: 'FINANCIAL_ADVISOR', label: 'Financial Advisor' },
      { value: 'TAX_PREPARER', label: 'Tax Preparer' },
      { value: 'TRANSLATOR', label: 'Translator/Interpreter' },
    ]
  },
  'CREATIVE': {
    label: 'Creative & Digital Services',
    icon: 'color-palette-outline',
    types: [
      { value: 'PHOTOGRAPHER', label: 'Photographer' },
      { value: 'VIDEOGRAPHER', label: 'Videographer' },
      { value: 'GRAPHIC_DESIGNER', label: 'Graphic Designer' },
      { value: 'WEB_DEVELOPER', label: 'Web Developer' },
      { value: 'APP_DEVELOPER', label: 'App Developer' },
      { value: 'CONTENT_CREATOR', label: 'Content Creator' },
      { value: 'SOCIAL_MEDIA_MANAGER', label: 'Social Media Manager' },
      { value: 'WRITER', label: 'Writer/Copywriter' },
      { value: 'MUSICIAN', label: 'Musician/DJ' },
      { value: 'ARTIST', label: 'Artist' },
    ]
  },
  'PERSONAL_CARE': {
    label: 'Personal Care & Wellness',
    icon: 'heart-outline',
    types: [
      { value: 'HAIRSTYLIST', label: 'Hairstylist/Barber' },
      { value: 'MAKEUP_ARTIST', label: 'Makeup Artist' },
      { value: 'NAIL_TECHNICIAN', label: 'Nail Technician' },
      { value: 'MASSAGE_THERAPIST', label: 'Massage Therapist' },
      { value: 'PERSONAL_TRAINER', label: 'Personal Trainer' },
      { value: 'YOGA_INSTRUCTOR', label: 'Yoga Instructor' },
      { value: 'NUTRITIONIST', label: 'Nutritionist' },
      { value: 'LIFE_COACH', label: 'Life Coach' },
    ]
  },
  'EDUCATION': {
    label: 'Education & Training',
    icon: 'school-outline',
    types: [
      { value: 'TUTOR', label: 'Private Tutor' },
      { value: 'LANGUAGE_TEACHER', label: 'Language Teacher' },
      { value: 'MUSIC_TEACHER', label: 'Music Teacher' },
      { value: 'DRIVING_INSTRUCTOR', label: 'Driving Instructor' },
      { value: 'SPORTS_COACH', label: 'Sports Coach' },
      { value: 'SKILLS_TRAINER', label: 'Skills Trainer' },
    ]
  },
  'SPECIALIZED': {
    label: 'Specialized Services',
    icon: 'construct-outline',
    types: [
      { value: 'TAILOR', label: 'Tailor/Seamstress' },
      { value: 'COBBLER', label: 'Cobbler/Shoe Repair' },
      { value: 'LOCKSMITH', label: 'Locksmith' },
      { value: 'SECURITY_GUARD', label: 'Security Guard' },
      { value: 'EVENT_PLANNER', label: 'Event Planner' },
      { value: 'CATERER', label: 'Caterer' },
      { value: 'TOUR_GUIDE', label: 'Tour Guide' },
      { value: 'PET_GROOMER', label: 'Pet Groomer' },
      { value: 'VETERINARY_ASSISTANT', label: 'Veterinary Assistant' },
    ]
  }
};

export const BUSINESS_CATEGORIES = {
  'RETAIL_COMMERCE': {
    label: 'Retail & Commerce',
    icon: 'storefront-outline',
    types: [
      { value: 'RETAIL_STORE', label: 'General Retail Store' },
      { value: 'GROCERY_MARKET', label: 'Grocery/Supermarket' },
      { value: 'FASHION_CLOTHING', label: 'Fashion & Clothing' },
      { value: 'ELECTRONICS_TECH', label: 'Electronics Store' },
      { value: 'HARDWARE_BUILDING', label: 'Hardware & Building Supplies' },
      { value: 'PHARMACY', label: 'Pharmacy' },
      { value: 'BOOKSTORE_STATIONERY', label: 'Bookstore & Stationery' },
      { value: 'ONLINE_STORE', label: 'E-commerce/Online Store' },
      { value: 'MOBILE_MONEY', label: 'Mobile Money Agent' },
      { value: 'FUEL_STATION', label: 'Fuel/Petrol Station' },
    ]
  },
  'FOOD_HOSPITALITY': {
    label: 'Food & Hospitality',
    icon: 'restaurant-outline',
    types: [
      { value: 'RESTAURANT_CAFE', label: 'Restaurant/Cafe' },
      { value: 'FAST_FOOD', label: 'Fast Food' },
      { value: 'BAKERY', label: 'Bakery' },
      { value: 'BAR_NIGHTCLUB', label: 'Bar/Nightclub' },
      { value: 'HOTEL_HOSPITALITY', label: 'Hotel/Lodge' },
      { value: 'GUEST_HOUSE', label: 'Guest House/B&B' },
      { value: 'CATERING_SERVICE', label: 'Catering Service' },
      { value: 'FOOD_DELIVERY', label: 'Food Delivery Service' },
    ]
  },
  'HEALTHCARE': {
    label: 'Healthcare & Wellness',
    icon: 'medical-outline',
    types: [
      { value: 'MEDICAL_DENTAL', label: 'Medical/Dental Clinic' },
      { value: 'VETERINARY', label: 'Veterinary Clinic' },
      { value: 'LABORATORY', label: 'Medical Laboratory' },
      { value: 'OPTICAL', label: 'Optical/Eye Care' },
      { value: 'PHYSIOTHERAPY', label: 'Physiotherapy Center' },
      { value: 'MENTAL_HEALTH', label: 'Mental Health Services' },
      { value: 'TRADITIONAL_MEDICINE', label: 'Traditional Medicine' },
    ]
  },
  'PROFESSIONAL_SERVICES': {
    label: 'Professional Services',
    icon: 'business-outline',
    types: [
      { value: 'CONSULTING_FIRM', label: 'Consulting Firm' },
      { value: 'LAW_FIRM', label: 'Law Firm' },
      { value: 'ACCOUNTING_FIRM', label: 'Accounting Firm' },
      { value: 'REAL_ESTATE', label: 'Real Estate Agency' },
      { value: 'INSURANCE_COMPANY', label: 'Insurance Services' },
      { value: 'FINANCIAL_SERVICES', label: 'Financial Services' },
      { value: 'MARKETING_AGENCY', label: 'Marketing/Advertising Agency' },
    ]
  },
  'MANUFACTURING': {
    label: 'Manufacturing & Production',
    icon: 'cog-outline',
    types: [
      { value: 'MANUFACTURING', label: 'Manufacturing' },
      { value: 'FOOD_PROCESSING', label: 'Food Processing' },
      { value: 'TEXTILE_GARMENT', label: 'Textile & Garment' },
      { value: 'FURNITURE_MAKING', label: 'Furniture Making' },
      { value: 'PRINTING_PUBLISHING', label: 'Printing & Publishing' },
      { value: 'PACKAGING', label: 'Packaging Services' },
    ]
  },
  'TECHNOLOGY': {
    label: 'Technology & Digital',
    icon: 'laptop-outline',
    types: [
      { value: 'SOFTWARE_COMPANY', label: 'Software Company' },
      { value: 'IT_SERVICES', label: 'IT Services & Support' },
      { value: 'TELECOM', label: 'Telecommunications' },
      { value: 'INTERNET_CAFE', label: 'Internet Cafe' },
      { value: 'COMPUTER_REPAIR', label: 'Computer/Phone Repair' },
      { value: 'WEB_HOSTING', label: 'Web Hosting/Domain Services' },
    ]
  },
  'CONSTRUCTION': {
    label: 'Construction & Real Estate',
    icon: 'home-outline',
    types: [
      { value: 'CONSTRUCTION', label: 'Construction Company' },
      { value: 'ARCHITECTURE', label: 'Architecture Firm' },
      { value: 'PROPERTY_MANAGEMENT', label: 'Property Management' },
      { value: 'INTERIOR_DESIGN', label: 'Interior Design' },
      { value: 'LANDSCAPING_COMPANY', label: 'Landscaping Company' },
    ]
  },
  'TRANSPORT_LOGISTICS': {
    label: 'Transport & Logistics',
    icon: 'airplane-outline',
    types: [
      { value: 'TRANSPORT_SERVICE', label: 'Transport Company' },
      { value: 'LOGISTICS_FREIGHT', label: 'Logistics & Freight' },
      { value: 'COURIER_COMPANY', label: 'Courier Company' },
      { value: 'CAR_RENTAL', label: 'Car Rental' },
      { value: 'TOUR_OPERATOR', label: 'Tour Operator' },
      { value: 'TRAVEL_AGENCY', label: 'Travel Agency' },
    ]
  },
  'EDUCATION_TRAINING': {
    label: 'Education & Training',
    icon: 'school-outline',
    types: [
      { value: 'SCHOOL', label: 'School/College' },
      { value: 'TRAINING_CENTER', label: 'Training Center' },
      { value: 'DAYCARE', label: 'Daycare/Nursery' },
      { value: 'DRIVING_SCHOOL', label: 'Driving School' },
      { value: 'LANGUAGE_SCHOOL', label: 'Language School' },
      { value: 'VOCATIONAL_TRAINING', label: 'Vocational Training' },
    ]
  },
  'AGRICULTURE': {
    label: 'Agriculture & Farming',
    icon: 'leaf-outline',
    types: [
      { value: 'FARM', label: 'Farm/Agricultural Business' },
      { value: 'LIVESTOCK', label: 'Livestock/Poultry' },
      { value: 'AGRO_PROCESSING', label: 'Agro-processing' },
      { value: 'AGRICULTURAL_SUPPLIES', label: 'Agricultural Supplies' },
      { value: 'FISHERY', label: 'Fishery/Aquaculture' },
    ]
  },
  'OTHER': {
    label: 'Other Services',
    icon: 'ellipsis-horizontal-outline',
    types: [
      { value: 'FITNESS_CENTER', label: 'Gym/Fitness Center' },
      { value: 'SALON_SPA', label: 'Beauty Salon/Spa' },
      { value: 'ENTERTAINMENT', label: 'Entertainment/Recreation' },
      { value: 'IMPORT_EXPORT', label: 'Import/Export' },
      { value: 'MINING_ENERGY', label: 'Mining/Energy' },
      { value: 'WASTE_MANAGEMENT', label: 'Waste Management' },
      { value: 'SECURITY_COMPANY', label: 'Security Company' },
      { value: 'CLEANING_COMPANY', label: 'Cleaning Company' },
      { value: 'EVENT_MANAGEMENT', label: 'Event Management' },
      { value: 'OTHER', label: 'Other' },
    ]
  }
};

export const REGISTRATION_STATUS = [
  { value: 'REGISTERED', label: 'Yes, fully registered' },
  { value: 'INFORMAL', label: 'No, operating informally' },
  { value: 'IN_PROCESS', label: 'Registration in process' },
];

export const VEHICLE_TYPES = [
  { value: 'bicycle', label: 'Bicycle', requiresLicense: false },
  { value: 'motorcycle', label: 'Motorcycle/Boda Boda', requiresLicense: true },
  { value: 'scooter', label: 'Scooter', requiresLicense: true },
  { value: 'car', label: 'Car', requiresLicense: true },
  { value: 'van', label: 'Van', requiresLicense: true },
  { value: 'truck', label: 'Truck', requiresLicense: true },
];

export const DELIVERY_CATEGORIES = [
  { value: 'food', label: 'Food & Beverages', icon: 'fast-food-outline' },
  { value: 'groceries', label: 'Groceries & Essentials', icon: 'basket-outline' },
  { value: 'packages', label: 'Packages & Documents', icon: 'cube-outline' },
  { value: 'medicine', label: 'Medicine & Healthcare', icon: 'medical-outline' },
  { value: 'electronics', label: 'Electronics & Gadgets', icon: 'phone-portrait-outline' },
  { value: 'clothing', label: 'Clothing & Fashion', icon: 'shirt-outline' },
  { value: 'fragile', label: 'Fragile Items', icon: 'warning-outline' },
  { value: 'heavy', label: 'Heavy/Large Items', icon: 'barbell-outline' },
  { value: 'all', label: 'All Categories', icon: 'checkmark-circle-outline' },
];

// Helper function to get all business types for backend
export const getAllBusinessTypes = () => {
  const types = [];
  
  // Add individual service provider types
  Object.values(INDIVIDUAL_CATEGORIES).forEach(category => {
    category.types.forEach(type => {
      types.push({
        value: type.value,
        label: type.label,
        category: 'INDIVIDUAL',
        requiresVehicle: type.requiresVehicle || false
      });
    });
  });
  
  // Add business types
  Object.values(BUSINESS_CATEGORIES).forEach(category => {
    category.types.forEach(type => {
      types.push({
        value: type.value,
        label: type.label,
        category: 'BUSINESS'
      });
    });
  });
  
  return types;
};

// Map to backend business types (from business_types.py)
export const mapToBackendType = (frontendType) => {
  const mapping = {
    // Individual service providers
    'COURIER': 'TRANSPORT_SERVICE',
    'TAXI_DRIVER': 'TRANSPORT_SERVICE',
    'TRUCK_DRIVER': 'TRANSPORT_SERVICE',
    'BODA_BODA': 'TRANSPORT_SERVICE',
    'PLUMBER': 'HOME_SERVICES',
    'ELECTRICIAN': 'HOME_SERVICES',
    'CARPENTER': 'HOME_SERVICES',
    'MECHANIC': 'AUTOMOTIVE_REPAIR',
    'CONSULTANT': 'PROFESSIONAL_SERVICES',
    'PHOTOGRAPHER': 'CREATIVE_SERVICES',
    'HAIRSTYLIST': 'SALON_SPA',
    'TUTOR': 'EDUCATION_TRAINING',
    
    // Businesses - direct mapping
    'RETAIL_STORE': 'RETAIL_STORE',
    'RESTAURANT_CAFE': 'RESTAURANT_CAFE',
    'MEDICAL_DENTAL': 'MEDICAL_DENTAL',
    'HOTEL_HOSPITALITY': 'HOTEL_HOSPITALITY',
    // ... add more mappings as needed
    
    // Default
    'OTHER': 'OTHER'
  };
  
  return mapping[frontendType] || 'OTHER';
};